File: packages/fxa-email-service/src/providers/mock.rs
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, you can obtain one at https://mozilla.org/MPL/2.0/.

use super::{Headers, Provider};
use app_errors::AppResult;

pub struct MockProvider;

impl Provider for MockProvider {
    fn send(
        &self,
        _to: &str,
        _cc: &[&str],
        _headers: Option<&Headers>,
        _subject: &str,
        _body_text: &str,
        _body_html: Option<&str>,
    ) -> AppResult<String> {
        Ok(String::from("deadbeef"))
    }
}

File: packages/fxa-settings/src/components/Settings/SubRow/mock.ts
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import AuthClient from 'fxa-auth-client';

/// mock localStorage state to allow MfaGuard to work in stories
export function initLocalAccount() {
  const uid = 'abc123';
  const accounts = {
    [uid]: {
      uid,
      sessionToken: 'sessionToken',
      email: 'user@example.com',
      verified: true,
      lastLogin: Date.now(),
    },
  };
  window.localStorage.setItem(
    `__fxa_storage.accounts`,
    JSON.stringify(accounts)
  );
  window.localStorage.setItem(
    `__fxa_storage.currentAccountUid`,
    JSON.stringify(uid)
  );
}

/// mock AuthClient to allow MfaGuard to work in stories
export const mockAuthClient = {
  mfaRequestOtp: async (st: string, sc: string) => {
    return undefined;
  },
  mfaOtpVerify: async (st: string, code: string, sc: string) => {
    return { accessToken: 'storybook-jwt' } as any;
  },
} as unknown as AuthClient;
